### PR TITLE
Fixes #12621: Explore alternative format for compliance table

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/12621

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/batch/AutomaticReportsCleaner.scala [hunk 1]
```diff
@@ -372,11 +374,19 @@ case class AutomaticReportsCleaning(
 
           case ActiveCleaner =>
             val now = DateTime.now
-            val target = now.minusDays(ttl)
-            val formattedDate = formatDate(target)
+            val reportsCommand = DeleteCommand.Reports(now.minusDays(reportsttl))
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/batch/AutomaticReportsCleaner.scala]: You shoudl do the same thing for reports command than compliance command 

 if (reportsttl > 0)  { Some( command) } else None

File: rudder-core/src/main/scala/com/normation/rudder/batch/AutomaticReportsCleaner.scala [hunk 2]
```diff
@@ -264,60 +265,61 @@ trait DatabaseCleanerActor extends LiftActor {
...
   //////////////////// implementation details ////////////////////
   ////////////////////////////////////////////////////////////////
 
-  private case class LADatabaseCleaner(cleanaction:CleanReportAction,ttl:Int) extends DatabaseCleanerActor with Loggable {
+  private case class LADatabaseCleaner(cleanaction:CleanReportAction, reportsttl:Int, compliancettl:Int) extends DatabaseCleanerActor with Loggable {
     updateManager =>
 
     private[this] val reportLogger = ReportLogger
-    private[this] val automatic = ttl > 0
+    private[this] val automatic = reportsttl > 0 || compliancettl > 0
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/batch/AutomaticReportsCleaner.scala]: Or you should change the logic so you only check that reportsTtl  > 0

File: rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ComplianceRepository.scala [hunk 3]
```diff
@@ -67,16 +72,29 @@ object RunCompliance {
...
+      , comp  => ComplianceLevel.unapply(comp).get
+    )
+  }
+
   /*
    * Save a list of node compliance reports
    */
   override def saveRunCompliance(reports: List[NodeStatusReport]): Box[List[NodeStatusReport]] = {
+
+    println(reports)
```
Comment (VinceMacBuche) [inline rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ComplianceRepository.scala]: to remove (or put in logger debug if you want) :)

File: rudder-core/src/test/scala/com/normation/rudder/db/GenerateCompliance.scala [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+import net.liftweb.common.EmptyBox
+import net.liftweb.common.Full
+import org.joda.time.DateTime
+import org.joda.time.Interval
+
+import scala.util.Random
+
+
+/*
+ * Allow to generate false compliance for testing purpose
```
Comment (VinceMacBuche) [inline rudder-core/src/test/scala/com/normation/rudder/db/GenerateCompliance.scala]: At last, A service for "All green" reporting mode ;)

**Comment by VinceMacBuche** [comment]
You have two commits, are you sure you cant to keep the first one ?

**Comment by VinceMacBuche** [comment]
Maybe it's interesting to keep it but i'll put a Ref #12621 at the beginning of the commit, explaining it's an experiment

File: rudder-core/src/test/scala/com/normation/rudder/db/GenerateCompliance.scala (hunk 4 above)
Comment (fanf) [reply rudder-core/src/test/scala/com/normation/rudder/db/GenerateCompliance.scala]: not even! it randomize output, so when you generate sufficiently big number of reports, you get an almost flat compliance

File: rudder-core/src/main/scala/com/normation/rudder/batch/AutomaticReportsCleaner.scala (hunk 1 above)
Comment (fanf) [reply rudder-core/src/main/scala/com/normation/rudder/batch/AutomaticReportsCleaner.scala]: yep

**Comment by fanf** [comment]
PR rebased

File: rudder-core/src/main/scala/com/normation/rudder/batch/AutomaticReportsCleaner.scala (hunk 1 above)
Comment (VinceMacBuche) [reply rudder-core/src/main/scala/com/normation/rudder/batch/AutomaticReportsCleaner.scala]: Ok so we changed the condtion to run the automatic cleaning so reports ttl is always > 0 , no need to check it here

**Comment by VinceMacBuche** [comment]
can you put a hash before the name of the first commit please ? so Redmine will like it to this issue automatically

**Comment by fanf** [comment]
Sorry, corrected

File: rudder-core/src/test/scala/com/normation/rudder/db/GenerateCompliance.scala (hunk 4 above)
Comment (VinceMacBuche) [reply rudder-core/src/test/scala/com/normation/rudder/db/GenerateCompliance.scala]: Yes i have seen but it's the first step to that audit mode needed by all Rudder users !!

**Comment by fanf** [comment]
OK, merging this PR

File: rudder-core/src/main/scala/com/normation/rudder/services/system/DatabaseManager.scala [hunk 5]
```diff
@@ -111,14 +121,14 @@ class DatabaseManagerImpl(
...
-     val deleteNodeConfigs = expectedReportsRepo.deleteNodeConfigurations(date) ?~! "An error occured while deleting Node Configurations"
-     val deleteNodeCompliances = expectedReportsRepo.deleteNodeCompliances(date) ?~! "An error occured while deleting Node Compliances"
+   def deleteEntries(reports: DeleteCommand.Reports, complianceLevels: Option[DeleteCommand.ComplianceLevel]) : Box[Int] = {
+     val nodeReports = reportsRepository.deleteEntries(reports.date) ?~! "An error occured while deleting reports"
+     val nodeConfigs = expectedReportsRepo.deleteNodeConfigIdInfo(reports.date) ?~! "An error occured while deleting old node configuration IDs"
+     val deleteNodeConfigs = expectedReportsRepo.deleteNodeConfigurations(reports.date) ?~! "An error occured while deleting Node Configurations"
+     val deleteNodeCompliances = expectedReportsRepo.deleteNodeCompliances(reports.date) ?~! "An error occured while deleting Node Compliances"
+     val deleteNodeComplianceLevels = Box(complianceLevels).flatMap(c => expectedReportsRepo.deleteNodeComplianceLevels(c.date) ?~! "An error occured while deleting Node Compliances")
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/rudder/services/system/DatabaseManager.scala:129]: This part fails when we disable deletion of complianceLevels as it converts to Empty

**Pull request merged**